src/app_service/service/transaction/service.nim

```diff
@@ -474,27 +472,22 @@ QtObject:
         finalPassword = common_utils.hashPassword(password)
     try:
       var chainID = 0
-      var isEthTx = false
 
       if(selectedRoutes.len > 0):
         chainID = selectedRoutes[0].fromNetwork.chainID
 
-      var tokenSymbol = ""
-      if self.isCollectiblesTransfer(sendType):
-        tokenSymbol = assetKey
-      else:
-        let token = self.tokenService.getTokenBySymbolByTokensKey(assetKey)
-        if token != nil:
-          tokenSymbol = token.symbol
-
+      let token = self.tokenService.getTokenBySymbolByTokensKey(assetKey)
+      if token != nil:
         let network = self.networkService.getNetwork(chainID)
-        if network.nativeCurrencySymbol == tokenSymbol:
-          isEthTx = true
-
-      if(isEthTx):
-        self.transferEth(fromAddr, toAddr, tokenSymbol, uuid, selectedRoutes, finalPassword)
+        if network.nativeCurrencySymbol == token.symbol:
+          self.transferEth(fromAddr, toAddr, token.symbol, uuid, selectedRoutes, finalPassword)
+          return
+        # else continue with token transfer
       else:
-        self.transferToken(fromAddr, toAddr, tokenSymbol, uuid, selectedRoutes, finalPassword, sendType, tokenName, isOwnerToken)
+        error "Token not found for", assetKey=assetKey
```

Comment:
this will always print the error for non ETH/ERC20 transfers, won't it?

Reply:
It seems so given that collectibles are not in the list. Fixing it right now.